File: cs_4234/projects/p1/hello.f
      implicit none
      include 'mpif.h'
      integer myid, numprocs, ierr

      call mpi_init( ierr )
      call mpi_comm_rank( mpi_comm_world, myid, ierr )
      call mpi_comm_size( mpi_comm_world, numprocs, ierr )

      print *, "hello from ", myid, " of ", numprocs

      call mpi_finalize( ierr )
      stop
      end

File: cs_4234/projects/p1/hello.c

#include "mpi.h"
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[])
{
   int myid, numprocs, namelen; 
   char processor_name[MPI_MAX_PROCESSOR_NAME];

   MPI_Init(&argc,&argv);
   MPI_Comm_size(MPI_COMM_WORLD,&numprocs);
   MPI_Comm_rank(MPI_COMM_WORLD,&myid);

   MPI_Get_processor_name(processor_name, &namelen);
   printf("hello from %s: process %d of %d\n", 
           processor_name, myid, numprocs);
 
   MPI_Finalize();
}
